src/main/java/mate/acadamy/internetshop/dao/impl/ItemDaoImpl.java

```diff
@@ -23,16 +35,43 @@ public Item get(Long id) {
 
     @Override
     public Item update(Item item) {
-        return null;
+        for (int i = 0; i < Storage.items.size(); i++) {
+            if (Storage.items.get(i).getId().equals(item.getId())) {
+                Storage.items.set(i, item);
+                return item;
+            }
+        }
+        throw new NoSuchElementException("Can`t find item " + item.getName());
     }
 
     @Override
-    public void delete(Long id) {
-
+    public Item delete(Long id) {
+        Optional<Item> currentItem = Storage.items
+                .stream()
+                .filter(x -> x.getId().equals(id))
+                .findAny();
+        if (!currentItem.equals(Optional.of(get(id)))) {
+            Item deletedItem = get(id);
+            Storage.items.removeIf(x -> x.getId().equals(id));
+            return deletedItem;
+        } else {
+            throw new NoSuchElementException("Can`t find item with id " + id);
+        }
     }
 
     @Override
-    public void delete(Item item) {
+    public Item delete(Item item) {
```

Comment:
we have duplicate code in this method and in delete(Long id) method

Reply:
Hear we search by item, there by id, the structure of checking look like same, but, have small difference. I make some changes in ```equals``` block

Reply:
Okey, i left only one method - ```delete(Long id)```